bin/did-modify

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+#
+# Copyright:: Copyright 2018 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+function usage() {
+  cat /dev/stdin <<DOC
+Usage: ${0##*/} [OPTIONS]
+
+Prints "true" to STDOUT if any files matching GLOBS were modified between HEAD and GITREF. Otherwise, prints "false".
+
+OPTIONS:
+    --git-ref=HEAD          A valid Git reference (e.g. HEAD, master, origin/master, etc). Defaults to "HEAD~1"
+    --globs=GLOB1,GLOB2     A list of glob patterns to inspect to determine if there are changes. Defaults to "*"
+DOC
+}
+
+function did_modify() {
+  globs="*"
+  git_ref="HEAD~1"
+
+  for opt in "$@"
+  do
+    case $opt in
+      --globs=*)
+        globs="${opt#*=}"
+        shift
```

Comment:
I'm not clear why there are `shift` calls. `$opt` should already be changing since the `for` loop is iterating over the values.

Reply:
Ah, it looks kind of like the paradigm I'm used to seeing with `getopts` parsing. I tried removing the `shift`s and I'm pretty sure they're superfluous, but if you see different behavior without them, I'd be very interested to understand why.